metaflow/metaflow_config.py

```diff
@@ -198,4 +201,28 @@ def get_authenticated_boto3_client(module, params={}):
             except requests.exceptions.HTTPError as e:
                 raise MetaflowException(repr(e))
         return boto3.session.Session(**cached_aws_sandbox_creds).client(module, **params)
+
+    if METAFLOW_AWS_ARN:
+        from datetime import datetime
+        from botocore.credentials import AssumeRoleCredentialFetcher, DeferredRefreshableCredentials
+        from botocore.session import Session
+        from dateutil.tz import tzlocal
+
+        source_session = boto3.Session()
+        # Use profile to fetch assume role credentials
+        fetcher = AssumeRoleCredentialFetcher(
+            client_creator=source_session._session.create_client,
+            source_credentials=source_session.get_credentials(),
+            role_arn=METAFLOW_AWS_ARN
+        )
+        botocore_session = Session()
+        botocore_session._credentials = DeferredRefreshableCredentials(
+            method="assume-role",
+            refresh_using=fetcher.fetch_credentials,
+            time_fetcher=lambda: datetime.now(tzlocal()),
+        )
+
+        session = boto3.Session(botocore_session=botocore_session, region_name="us-west-2")
```

Comment:
`region_name` should be an environment variable..

Reply:
Done